### ISSUE TITLE
File attaching at processing state

### ISSUE THREAD

**Issue by zerg11** [mattermost/mattermost#24363]
Hello everyone,
We faced with strange behaviour of MM file attaching in self-hosted environment. Users in one time (it happens absolutely random) just can't load their attachements due to "processing" state in main window. When you cancel and re-attach - it won't help, and only restarting the server (docker image in our case) is actually helping. 
Logs (Problem is ~1:10pmUTC) and some screens from browser and client are in attach. Resources of the server (CPU, RAM, Network)  is in normal consumption.
When we open a console from browser in that mome
[logs.txt](https://github.com/mattermost/mattermost/files/12430836/logs.txt)
nt - it just says that POST 200 OK, but not genereting a thumbnail message as usual.
We faced this in 7.x.x version and 8.0.0 as well.

Any help with that behavior appriciated!

Server OS: Ubuntu 22.04.2 LTS (8 Gb RAM and 4 vCPU).
Mattermost Docker Image: 7.10.0, 8.0.0
Mattermos Windows/Mac client: 5.4.0
Plugins installed: Autolink 1.4.0, Jitsi 2.0.1, Matterpoll 1.6.1, Remind 1.1.0.

#### Summary
Fall off file attaching periodically.

#### Steps to reproduce
Upload file as usual and getting infinite "processing" state without any errors.

#### Expected behavior
Upload file as usual and getting preview.
<img width="1440" alt="pic_application" src="https://github.com/mattermost/mattermost/assets/70346681/170565a8-3c75-4321-9a56-68c62ab09b4b">
<img width="1731" alt="pic_browser_console" src="https://github.com/mattermost/mattermost/assets/70346681/3b8afed8-d6b8-435a-8fa0-baea9fa1f6af">
<img width="1731" alt="pic_browser_consoe_2" src="https://github.com/mattermost/mattermost/assets/70346681/c424f77a-65fd-4a69-b00d-b17b7def9a2b">

**Comment by zerg11** [comment]
[logs.txt](https://github.com/mattermost/mattermost/files/12430855/logs.txt)

**Comment by beeritis** [comment]
This is still an issue , using Mattermost version  9.2.2 on Redhat 8.9 . 

One thing of note is that the files were actually being written into  /opt/mattermost/data/<ddmmdd>/teams/noteam/channels/<id> etc. . So the files had been uploaded succesfully but from the client side it was still displaying as "processing" 

A restart of the mattermost service resolved the issue in the meantime.

**Comment by iox** [comment]
We are affected as well with 9.2.2, with an S3 backend (DigitalOcean spaces). After a few days of usage, file uploads stop working, and only a restart of the mattermost service fixes it.

Looking at the S3 storage, the files appear correctly in the bucket:

![Screenshot_20240117_112647](https://github.com/mattermost/mattermost/assets/452823/282abee5-8868-488f-9ca5-8c463472ee4e)

But the client never gets a response from the server. Looking at the browser inspector, the request times out after 60 seconds.

**Comment by eli-yip** [comment]
Same problem in 9.5.0 and 9.4.1.

**Comment by devinbinnie** [comment]
Is anyone here able to provide logs for both the client and server when this issue reproduces? I'm curious where the failure point since it sounds from the above that the files are successfully uploading, but the response back to the client is failing. Thanks!

**Comment by zerg11** [comment]
@devinbinnie
Here we go for a logs, happened at 11:18. Just all clients were at "processing" state, when uploads pic. After restarting container with MM, goes well again. Version: 9.2.2
But nothing I could point to.
<img width="899" alt="screen_files" src="https://github.com/mattermost/mattermost/assets/70346681/759b3251-86d6-4719-9963-66192c973850">
[logs.txt](https://github.com/mattermost/mattermost/files/14277144/logs.txt)
[client_logs.txt](https://github.com/mattermost/mattermost/files/14277148/client_logs.txt)

**Comment by devinbinnie** [comment]
@zerg11 Yeah looks like something is getting stuck, since it doesn't seem like there's any error being returned from the server. 

I do see a couple lines that raise a question: is it always the same file that triggers this, or can it be any file at all?

If not, next steps here would be to turn on debug logging on the server so that we can trace the API call and see if anything else is being thrown from a different server.

**Comment by zerg11** [comment]
@devinbinnie Yeah, we tested with a randomly images, even with big sizes, resolutions, name scopes ("+""-_-", etc). Nothing was stucked, everything as expected. But we have a lot of users, so it's hard to tell if they triggers this behavior or not.

**Comment by lieut-data** [comment]
Hi everyone! Reviewing this issue and there could be multiple problems at play. Unfortunately, we don't have enough information to reproduce this as described.

If folks are still seeing this issue, could you try reproducing on our community instance and share reproduction steps? For now, I'm going to close this issue until we have something on which to make progress.
